src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1450,6 +1475,13 @@ public String getNextScriptName(String path) {
         return shell.generateScriptName().replaceFirst("[.]groovy$", "");
     }
 
+    /** Has the execution been marked done - note that legacy builds may not have that flag persisted, in which case
+     *  we look for a single FlowEndNode head (see: {@link #isComplete()} and {@link FlowExecution#isComplete()})
+     */
+    public boolean isDoneFlagSet() {
```

Comment:
Not simply `isDone()`?

Reply:
Nope, we have an isComplete and this is only checking for the flag itself (which may not be 100% trustworthy on its own)